module/zfs/dmu_objset.c

```diff
@@ -1072,6 +1079,58 @@ dmu_objset_snapshot_one(const char *fsname, const char *snapname)
 	return (err);
 }
 
+typedef struct dmu_objset_upgrade_args {
+	objset_t *dua_os;
+	int (*dua_cb)(objset_t *);
+} dmu_objset_upgrade_arg_t;
+
+static void
+dmu_objset_upgrade_trampoline(void *data)
```

Comment:
[style] The existing convention would be to name this function, `dmu_objset_upgrade_thread()`.